validation/start.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/mndrix/tap-go"
+	rspecs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validation/util"
+	uuid "github.com/satori/go.uuid"
+)
+
+func wrap(t *tap.T, expected bool, specErr error, detailedErr error) {
+	t.Ok(expected, specErr.(*specerror.Error).Err.Err.Error())
+	diagnostic := map[string]string{
+		"reference": specErr.(*specerror.Error).Err.Reference,
+	}
+
+	if detailedErr != nil {
+		diagnostic["error"] = detailedErr.Error()
+		if e, ok := detailedErr.(*exec.ExitError); ok {
+			if len(e.Stderr) > 0 {
+				diagnostic["stderr"] = string(e.Stderr)
+			}
+		}
+	}
+	t.YAML(diagnostic)
+}
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+
+	bundleDir, err := util.PrepareBundle()
+	if err != nil {
+		util.Fatal(err)
+	}
+	defer os.RemoveAll(bundleDir)
+
+	containerID := uuid.NewV4().String()
+
+	r, err := util.NewRuntime(util.RuntimeCommand, bundleDir)
+	if err != nil {
+		util.Fatal(err)
+	}
+	g := util.GetDefaultGenerator()
+	g.SetProcessArgs([]string{"sh", "-c", fmt.Sprintf("echo 'process called' >> %s", "/output")})
+	r.SetConfig(g)
+	output := filepath.Join(bundleDir, g.Spec().Root.Path, "output")
+
+	// start without id
+	err = r.Start()
+	wrap(t, err != nil, specerror.NewError(specerror.StartWithoutIDGenError, fmt.Errorf("start` operation MUST generate an error if it is not provided the container ID"), rspecs.Version), err)
+
+	// set id for the remaining tests
+	r.SetID(containerID)
+
+	// start a not `created` container - case one: non-exist container
+	err = r.Start()
+	wrap(t, err != nil, specerror.NewError(specerror.StartNotCreatedGenError, fmt.Errorf("attempting to `start` a container that is not `created` MUST generate an error"), rspecs.Version), err)
+
+	r.Create()
+	// start a `created` container
+	err = r.Start()
+	outputData, _ := ioutil.ReadFile(output)
+	// check the output
+	wrap(t, err == nil && string(outputData) == "process called\n", specerror.NewError(specerror.StartProcImplement, fmt.Errorf("`start` operation MUST run the user-specified program as specified by `process`"), rspecs.Version), err)
+
+	// start a not `created` container - case two: exist and `stopped`
+	util.WaitingForStatus(r, util.LifecycleStatusStopped, time.Second*10, time.Second*1)
+	err = r.Start()
+	// must generate an error
+	wrap(t, err != nil, specerror.NewError(specerror.StartNotCreatedGenError, fmt.Errorf("attempting to `start` a container that is not `created` MUST generate an error"), rspecs.Version), err)
+
+	outputData, _ = ioutil.ReadFile(output)
```

Comment:
You shouldn't be ignoring the returned error here (or in your other `ReadFile` calls).  If one of those errors, you can die with a non-zero exit.  node-tap will complain about that:

```console
$ cat validation/test 
#!/bin/sh
cat <<EOF
TAP version 13
ok 1 - ok
EOF
exit 1
$ tap ./validation/test 
./validation/test ..................................... 1/2
  not ok ./validation/test
    timeout: 30000
    file: ./validation/test
    command: ./validation/test
    args: []
    stdio:
      - 0
      - pipe
      - 2
    cwd: /home/wking/.local/lib/go/src/github.com/opencontainers/runtime-tools
    failures:
      - tapError: no plan
    exitCode: 1
…
```